### PR TITLE
New package: hwloc

### PR DESCRIPTION
**Pull request by 3rav**
Add hwloc, base on: https://www.archlinux.org/packages/extra/x86_64/hwloc/

### CODE REVIEW

File: mingw-w64-hwloc/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+prepare() {
+  cd hwloc-${pkgver}
+  autoreconf -fiv
+}
+
+build() {
+  cd hwloc-${pkgver}
+  ./configure \
+    --prefix=/usr \
```
Comment (lazka) [inline mingw-w64-hwloc/PKGBUILD:32]: These paths are all not mingw paths. Did you test the package?

Comment (3rav) [reply mingw-w64-hwloc/PKGBUILD:32]: If I have:

**make check and --enable-plugins:**
```
FAIL: test-fake-plugin.sh
PASS: test-hwloc-diffpatch.sh
PASS: test-hwloc-info.sh
PASS: test-hwloc-distrib.sh
PASS: test-hwloc-annotate.sh
PASS: test-hwloc-calc.sh
PASS: test-hwloc-compress-dir.sh
============================================================================
Testsuite summary for hwloc 2.2.0
============================================================================
# TOTAL: 7
# PASS:  6
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See utils/hwloc/test-suite.log
Please report to http://github.com/open-mpi/hwloc/issues
============================================================================
make[5]: *** [Makefile:1342: test-suite.log] Błąd 1
make[5]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/hwloc'
make[4]: *** [Makefile:1450: check-TESTS] Błąd 2
make[4]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/hwloc'
make[3]: *** [Makefile:1590: check-am] Błąd 2
make[3]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/hwloc'
make[2]: *** [Makefile:1234: check-recursive] Błąd 1
make[2]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/hwloc'
make[1]: *** [Makefile:461: check-recursive] Błąd 1
make[1]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils'
make: *** [Makefile:639: check-recursive] Błąd 1
```

**only make check:**
```
PASS: test-hwloc-diffpatch.sh
PASS: test-hwloc-info.sh
PASS: test-hwloc-distrib.sh
PASS: test-hwloc-annotate.sh
PASS: test-hwloc-calc.sh
PASS: test-hwloc-compress-dir.sh
============================================================================
Testsuite summary for hwloc 2.2.0
============================================================================
# TOTAL: 6
# PASS:  6
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
make[5]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/hwloc'
make[4]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/hwloc'
make[3]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/hwloc'
make[2]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/hwloc'
Making check in lstopo
make[2]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/lstopo'
make  check-TESTS
make[3]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/lstopo'
make[4]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/lstopo'
PASS: test-lstopo.sh
============================================================================
Testsuite summary for hwloc 2.2.0
============================================================================
# TOTAL: 1
# PASS:  1
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
make[4]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/lstopo'
make[3]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/lstopo'
make[2]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils/lstopo'
make[2]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils'
make[2]: Nie ma nic do zrobienia w 'check-am'.
make[2]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils'
make[1]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/utils'
Making check in tests
make[1]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests'
Making check in hwloc
make[2]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc'
Making check in .
make[3]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc'
make  hwloc_api_version.exe hwloc_list_components.exe hwloc_bitmap.exe hwloc_bitmap_string.exe hwloc_bitmap_compare_inclusion.exe hwloc_get_closest_objs.exe hwloc_get_obj_covering_cpuset.exe hwloc_get_cache_covering_cpuset.exe hwloc_get_largest_objs_inside_cpuset.exe hwloc_get_next_obj_covering_cpuset.exe hwloc_get_obj_inside_cpuset.exe hwloc_get_shared_cache_covering_obj.exe hwloc_get_obj_below_array_by_type.exe hwloc_bitmap_first_last_weight.exe hwloc_bitmap_singlify.exe hwloc_type_depth.exe hwloc_type_sscanf.exe hwloc_bind.exe hwloc_get_last_cpu_location.exe hwloc_get_area_memlocation.exe hwloc_object_userdata.exe hwloc_synthetic.exe hwloc_backends.exe hwloc_pci_backend.exe hwloc_is_thissystem.exe hwloc_distances.exe hwloc_groups.exe hwloc_insert_misc.exe hwloc_topology_allow.exe hwloc_topology_restrict.exe hwloc_topology_dup.exe hwloc_topology_diff.exe hwloc_topology_abi.exe hwloc_obj_infos.exe hwloc_iodevs.exe cpuset_nodeset.exe xmlbuffer.exe gl.exe
make[4]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc'
  CCLD     hwloc_api_version.exe
  CCLD     hwloc_list_components.exe
  CCLD     hwloc_bitmap.exe
  CCLD     hwloc_bitmap_string.exe
  CC       hwloc_bitmap_compare_inclusion.o
  CCLD     hwloc_get_closest_objs.exe
  CCLD     hwloc_get_obj_covering_cpuset.exe
  CCLD     hwloc_get_cache_covering_cpuset.exe
  CCLD     hwloc_get_largest_objs_inside_cpuset.exe
  CCLD     hwloc_get_next_obj_covering_cpuset.exe
  CCLD     hwloc_get_obj_inside_cpuset.exe
  CCLD     hwloc_get_shared_cache_covering_obj.exe
  CCLD     hwloc_get_obj_below_array_by_type.exe
  CCLD     hwloc_bitmap_first_last_weight.exe
  CCLD     hwloc_bitmap_singlify.exe
  CC       hwloc_type_depth.o
  CC       hwloc_type_sscanf.o
  CC       hwloc_bind.o
  CCLD     hwloc_get_last_cpu_location.exe
  CCLD     hwloc_get_area_memlocation.exe
  CCLD     hwloc_synthetic.exe
  CCLD     hwloc_object_userdata.exe
  CC       hwloc_backends.o
  CCLD     hwloc_pci_backend.exe
  CCLD     hwloc_is_thissystem.exe
  CCLD     hwloc_distances.exe
  CCLD     hwloc_groups.exe
  CCLD     hwloc_insert_misc.exe
  CCLD     hwloc_topology_allow.exe
  CCLD     hwloc_topology_restrict.exe
  CCLD     hwloc_topology_dup.exe
  CCLD     hwloc_topology_diff.exe
  CC       hwloc_topology_abi.o
  CC       hwloc_iodevs.o
  CCLD     hwloc_obj_infos.exe
  CCLD     cpuset_nodeset.exe
  CCLD     xmlbuffer.exe
  CCLD     gl.exe
  CCLD     hwloc_bitmap_compare_inclusion.exe
  CCLD     hwloc_type_depth.exe
  CCLD     hwloc_type_sscanf.exe
  CCLD     hwloc_bind.exe
  CCLD     hwloc_backends.exe
  CCLD     hwloc_topology_abi.exe
  CCLD     hwloc_iodevs.exe
make[4]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc'
make  check-TESTS
make[4]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc'
make[5]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc'
PASS: hwloc_list_components.exe
PASS: hwloc_api_version.exe
PASS: hwloc_bitmap.exe
PASS: hwloc_bitmap_compare_inclusion.exe
PASS: hwloc_get_closest_objs.exe
PASS: hwloc_bitmap_string.exe
PASS: hwloc_get_cache_covering_cpuset.exe
PASS: hwloc_get_largest_objs_inside_cpuset.exe
PASS: hwloc_get_obj_covering_cpuset.exe
PASS: hwloc_get_obj_inside_cpuset.exe
PASS: hwloc_get_next_obj_covering_cpuset.exe
PASS: hwloc_get_obj_below_array_by_type.exe
PASS: hwloc_get_shared_cache_covering_obj.exe
PASS: hwloc_bitmap_first_last_weight.exe
PASS: hwloc_bitmap_singlify.exe
PASS: hwloc_type_depth.exe
PASS: hwloc_type_sscanf.exe
PASS: hwloc_bind.exe
PASS: hwloc_get_area_memlocation.exe
PASS: hwloc_get_last_cpu_location.exe
PASS: hwloc_synthetic.exe
PASS: hwloc_pci_backend.exe
PASS: hwloc_is_thissystem.exe
PASS: hwloc_backends.exe
PASS: hwloc_object_userdata.exe
PASS: hwloc_groups.exe
PASS: hwloc_distances.exe
PASS: hwloc_insert_misc.exe
PASS: hwloc_topology_restrict.exe
PASS: hwloc_topology_allow.exe
PASS: hwloc_topology_dup.exe
PASS: hwloc_topology_diff.exe
PASS: hwloc_obj_infos.exe
PASS: hwloc_topology_abi.exe
PASS: hwloc_iodevs.exe
PASS: cpuset_nodeset.exe
PASS: gl.exe
PASS: xmlbuffer.exe
============================================================================
Testsuite summary for hwloc 2.2.0
============================================================================
# TOTAL: 38
# PASS:  38
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
make[5]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc'
make[4]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc'
make[3]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc'
Making check in ports
make[3]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc/ports'
make   \

make[4]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc/ports'
make[4]: Nie ma nic do zrobienia w 'all'.
make[4]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc/ports'
make[3]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc/ports'
Making check in xml
make[3]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc/xml'
make  check-TESTS
make[4]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc/xml'
make[5]: Wejście do katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc/xml'
FAIL: 16intel64-manyVFs.xml
PASS: 16amd64-4distances.console.output
PASS: 16em64t-4s2c2t.console.output
PASS: 16-2gr2gr2n2c+misc.console.output
PASS: 16amd64-4distances.xml
PASS: 16em64t-4s2c2t-offlines.xml
PASS: 16em64t-4s2c2t.xml
PASS: 16-2gr2gr2n2c+misc.xml
PASS: 16amd64-8n2c-cpusets.xml
FAIL: 24em64t-2n6c2t-pci.xml
PASS: 16intel64-manyVFs.console.output
PASS: 16intel64-manyVFs.console.nocollapse.output
FAIL: 32em64t-2n8c2t-pci-normalio.xml
FAIL: 32em64t-2n8c2t-pci-wholeio.xml
FAIL: 64intel64-3g2n+2n-irregulargroups+pci.xml
PASS: 64intel64-3g2n+2n-irregulargroups+pci.console.output
PASS: 32em64t-2n8c2t-pci-noio.xml
FAIL: 96em64t-4n4d3ca2co-pci.xml
PASS: 8intel64-fakeKNL-A2A-hybrid.rootattachednumas.xml
FAIL: 192em64t-12gr2n8c2t.xml
FAIL: 192em64t-24n8c2t.xml
FAIL: 8em64t-2p2ca2co-nonodesets.v1tov2.xml
PASS: 64intel64-fakeKNL-SNC4-hybrid.xml
FAIL: 8ia64-2n2s2c+1n.v1tov2.xml
PASS: fakeheterodistances.xml
FAIL: 16amd64-4distances.v2tov1.xml
PASS: 16amd64-4distances.v1tov2.xml
FAIL: 28intel64-2p2g7c-CoDgroups.v1tov2.xml
FAIL: 28intel64-2p2g7c-CoD.nogroups.v1tov2.xml
PASS: 2intel64-1n2c-numaroot.v1tov2.xml
PASS: 8intel64-fakeKNL-A2A-hybrid.rootattachednumas.v1tov2.xml
PASS: 8intel64-fakeKNL-A2A-hybrid.rootattachednumas.v2tov1.xml
PASS: 64intel64-fakeKNL-SNC4-hybrid.v1tov2.xml
PASS: 64intel64-fakeKNL-SNC4-hybrid.v2tov1.xml
============================================================================
Testsuite summary for hwloc 2.2.0
============================================================================
# TOTAL: 34
# PASS:  21
# SKIP:  0
# XFAIL: 0
# FAIL:  13
# XPASS: 0
# ERROR: 0
============================================================================
See tests/hwloc/xml/test-suite.log
Please report to http://github.com/open-mpi/hwloc/issues
============================================================================
make[5]: *** [Makefile:719: test-suite.log] Błąd 1
make[5]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc/xml'
make[4]: *** [Makefile:827: check-TESTS] Błąd 2
make[4]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc/xml'
make[3]: *** [Makefile:1131: check-am] Błąd 2
make[3]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc/xml'
make[2]: *** [Makefile:1368: check-recursive] Błąd 1
make[2]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests/hwloc'
make[1]: *** [Makefile:455: check-recursive] Błąd 1
make[1]: Opuszczenie katalogu '/usr/local/Hwloc/src/hwloc-2.2.0/tests'
make: *** [Makefile:639: check-recursive] Błąd 1
```

**make check-am - no errors.**
from Makefile:
```
	       exit 1; } >&2
check-am: all-am
check: check-recursive
all-am: Makefile $(DATA)
```

Comment (lazka) [reply mingw-w64-hwloc/PKGBUILD:32]: Have a look at https://github.com/msys2/MINGW-packages/blob/master/mingw-w64-PKGBUILD-templates/PKGBUILD.autotools-tarball all paths need to be under `$MINGW_PREFIX`

**Comment by lazka** [comment]
Thanks!

**Pull request merged**